src/native/eventpipe/ep-types-forward.h

```diff
@@ -60,7 +61,7 @@ typedef struct _StreamWriterVtable StreamWriterVtable;
 
 #define EP_ACTIVITY_ID_SIZE EP_GUID_SIZE
 
-#define EP_MAX_STACK_DEPTH 100
+#define EP_MAX_STACK_DEPTH 512
```

Comment:
I believe there are 2 on the stack for every event write:
1. in ep_buffer_manager_write_event https://github.com/josalem/runtime/blob/b68132efc1f7bcb1d981f6c66e2fa9da566054dc/src/native/eventpipe/ep-buffer-manager.c#L943-L943
2. ep_buffer_write_event https://github.com/josalem/runtime/blob/b68132efc1f7bcb1d981f6c66e2fa9da566054dc/src/native/eventpipe/ep-buffer.c#L86-L86

It's always struck me as odd that we do the stack check twice. We can probably remove one of them. There are any uses of `ep_buffer_write_event` outside `EventPipeBufferManager` anyway.

On that note, is 512 too large? I set that value intending to have a conversation about what the limit _should_ be. I'd be content lowering it.

Reply:
I heard that 200 should be "enough", coming from a large UI library implementor. Alternative, use dynamic stack allocation and we could get this value from an env variable instead of hardcoded, but still bound it into sane limits.

Reply:
Tying it to a config flag is an interesting idea. That lets us use a smaller default value for the general case and gives a release valve to customers that need it. I think we'd need to move the struct off the stack to do that though since the size wouldn't be known till runtime. A `threadlocal static EventPipeStackContents *` would make it so we only ever make one. 

I think we should make this a second PR though. For now, let's pull the max count down to 200 and we can add the config flag after.

Reply:
Sorry I was meaning to review today and got derailed. For this part specifically though how about we leave it at 100 for this PR and make it configurable/change the default in a later PR? I'm not attached to this default value but everything else in this PR is a size improvement whereas this is a potential trace size/time regression for anyone who has large stacks. I would put it in an isolated PR with some discussion about why we think the new default is better than the old default and maybe we'll get some feedback from users on it.

Reply:
Works for me! I'll spin an issue/PR out of this comment chain, and we can go from there. It seems worthwhile to have a release valve for customers with large stacks. I would rather the default stay at 100 and we come back later to make it configurable, as you said.